### PR TITLE
Add documentation for restoration

### LINKED ISSUE #186: Document how to restore from backups
**Need:**
Create a how-to document to restore the data from backups 

**Motivation:**
This document will act as a guide for administrators who monitor the clusters and trigger the restore action if required

### PR DESCRIPTION
**Pull request by shreyas-s-rao**
Signed-off-by: Shreyas Rao <shreyas.sriganesh.rao@sap.com>

**What this PR does / why we need it**:
This PR adds documentation for the restoration process and workflow.

**Which issue(s) this PR fixes**:
Fixes #186 

**Special notes for your reviewer**:

**Release note**:

```improvement operator
Added documentation for restoration.
```

### CODE REVIEW

File: chart/etcd-backup-restore/values.yaml [hunk 1]
```diff
@@ -90,6 +90,8 @@ backup:
   #   accessKeySecret: secret-access-key-with-object-storage-privileges
   #   accessKeyID: access-key-id-with-object-storage-privileges
 
+metrics: basic
```
Comment (swapnilgm) [inline chart/etcd-backup-restore/values.yaml]: Do we really need this? Please don't add it unless there is  some strong use case. I preferred not to expose etcd configuration unless there is use case(specific requirement) for allowing that field configurable here. That's why i ignored many flags from https://github.com/etcd-io/etcd/blob/master/Documentation/op-guide/configuration.md.

File: doc/proposals/restoration.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+Restorer fetches and restores from the latest full snapshot present in the snapstore, and then fetches and applies delta snapshots on top of it. These snapshots are named `etcd-main/v1/Backup-xxxxxxxxxx/Full-aaaaaaaa-bbbbbbbb-xxxxxxxxxx`, `etcd-main/v1/Backup-xxxxxxxxxx/Incr-bbbbbbbb-cccccccc-yyyyyyyyyy` and so on, where `xxxxxxxxxx`,`yyyyyyyyyy` are UTC timestamps and `aaaaaaaa`,`bbbbbbbb`,`cccccccc` are etcd revision numbers. Each snapshot contains etcd events between two revisions as specified in the snapshot name. :warning: If in doubt about the validity of the etcd data, operators must first visually confirm any inconsistency in revision numbers between the latest snapshot set in the object store and the running etcd instance using the `etcdctl` tool before deciding to perform a manual restoration.
```
Comment (swapnilgm) [inline doc/proposals/restoration.md]:
```suggestion
Restorer fetches the latest full snapshot present in the snapstore  and restores from it, and then fetches the depending delta snapshots and applies on top of restored data directory. These snapshots are named `etcd-main/v1/Backup-xxxxxxxxxx/Full-aaaaaaaa-bbbbbbbb-xxxxxxxxxx`, `etcd-main/v1/Backup-xxxxxxxxxx/Incr-bbbbbbbb-cccccccc-yyyyyyyyyy` and so on, where `xxxxxxxxxx`,`yyyyyyyyyy` are UTC timestamps and `aaaaaaaa`,`bbbbbbbb`,`cccccccc` are etcd revision numbers. Each snapshot contains etcd events between two revisions as specified in the snapshot name. :warning: If in doubt about the validity of the etcd data, operators must first visually confirm any inconsistency in revision numbers between the latest snapshot set in the object store and the running etcd instance using the `etcdctl` tool before deciding to perform a manual restoration.
```

File: doc/proposals/restoration.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# Etcd data restoration
```
Comment (swapnilgm) [inline doc/proposals/restoration.md]: Add reference to this doc in [this section](https://github.com/gardener/etcd-backup-restore/blob/master/doc/usage/getting_started.md#etcd-data-directory-initialization) as well.

File: chart/etcd-backup-restore/values.yaml (hunk 1 above)
Comment (shreyas-s-rao) [reply chart/etcd-backup-restore/values.yaml]: This metric is exposed and set in the Gardener etcd chart at https://github.com/gardener/gardener/blob/17354845eeaae0461f38f95639301d77f6f16e69/charts/seed-controlplane/charts/etcd/values.yaml#L17. So I felt it's necessary to expose this in etcd-backup-restore chart as well.

Comment (swapnilgm) [reply chart/etcd-backup-restore/values.yaml]: Cool 👍

File: doc/proposals/restoration.md (hunk 3 above)
Comment (shreyas-s-rao) [reply doc/proposals/restoration.md]: Done

File: doc/proposals/restoration.md (hunk 2 above)
Comment (shreyas-s-rao) [reply doc/proposals/restoration.md]: Done

**Comment by shreyas-s-rao** [comment]
@swapnilgm Thanks for your review. I've addressed the suggestions. PTAL.

**Pull request merged**